### DISCUSSION TITLE
cannot login from client kali linux

### DISCUSSION THREAD

**Discussion by Chwbecca** [ltsp/ltsp#588] [Q&A] [answered]
i have the following problem: i have successfully created a kali linux client system. However at login, although I am sure that you are using the ltsp server users the login fails with the following error:

Oct 21 10:00:24 host510 sshd[28050]: debug1: PAM: initializing for "XXX"
Oct 21 10:00:24 host510 sshd[28050]: debug1: PAM: setting PAM_RHOST to "192.168.XXX.XXX"
Oct 21 10:00:24 host510 sshd[28050]: debug1: PAM: setting PAM_TTY to "ssh"
Oct 21 10:00:24 host510 sshd[28050]: debug2: monitor_read: 100 used once, disabling now
Oct 21 10:00:24 host510 sshd[28050]: debug3: mm_request_receive entering
Oct 21 10:00:24 host510 sshd[28050]: debug3: monitor_read: checking request 4
Oct 21 10:00:24 host510 sshd[28050]: debug3: mm_answer_authserv: service=ssh-connection, style=, role=
Oct 21 10:00:24 host510 sshd[28050]: debug2: monitor_read: 4 used once, disabling now
Oct 21 10:00:24 host510 sshd[28050]: debug3: userauth_finish: failure partial=0 next methods="publickey,password" [preauth]
Oct 21 10:00:24 host510 sshd[28050]: debug3: send packet: type 51 [preauth]
Oct 21 10:00:24 host510 sshd[28050]: debug3: receive packet: type 50 [preauth]
Oct 21 10:00:24 host510 sshd[28050]: debug1: userauth-request for user XXX service ssh-connection method password [preauth]
Oct 21 10:00:24 host510 sshd[28050]: debug1: attempt 1 failures 0 [preauth]
Oct 21 10:00:24 host510 sshd[28050]: debug2: input_userauth_request: try method password [preauth]
Oct 21 10:00:24 host510 sshd[28050]: debug3: mm_auth_password entering [preauth]
Oct 21 10:00:24 host510 sshd[28050]: debug3: mm_request_send entering: type 12 [preauth]
Oct 21 10:00:24 host510 sshd[28050]: debug3: mm_auth_password: waiting for MONITOR_ANS_AUTHPASSWORD [preauth]
Oct 21 10:00:24 host510 sshd[28050]: debug3: mm_request_receive_expect entering: type 13 [preauth]
Oct 21 10:00:24 host510 sshd[28050]: debug3: mm_request_receive entering [preauth]
Oct 21 10:00:24 host510 sshd[28050]: debug3: mm_request_receive entering
Oct 21 10:00:24 host510 sshd[28050]: debug3: monitor_read: checking request 12
Oct 21 10:00:24 host510 sshd[28050]: debug3: mm_answer_authpassword: sending result 0
Oct 21 10:00:24 host510 sshd[28050]: debug3: mm_request_send entering: type 13
Oct 21 10:00:24 host510 sshd[28050]: Failed none for XXX from 192.168.XXX.XXX port 56352 ssh2
Oct 21 10:00:24 host510 sshd[28050]: debug3: mm_auth_password: user not authenticated [preauth]
Oct 21 10:00:24 host510 sshd[28050]: debug3: user_specific_delay: user specific delay 0.000ms [preauth]
Oct 21 10:00:24 host510 sshd[28050]: debug3: ensure_minimum_time_since: elapsed 0.426ms, delaying 6.366ms

**Comment by alkisg** [comment]
1. In the LTSP client, what is the output of: `sudo grep username /etc/{passwd,group,shadow,gshadow}`. Replace username with the user that's trying to login. There shouldn't be any secrets there; if you want to hide the username, replace it in the output with the word "username".
2. If you press Ctrl+Alt+F2 and try to login, what's the error message?
3. Do you have ltsp and sshfs installed in the LTSP client?
4. If you run `ssh username@server` from the LTSP client, does it work?

To get a root shell in the LTSP client, see the last example of https://ltsp.org/man/ltsp.conf/

**Comment by Chwbecca** [comment]
Entering with the root user locally I tried to ssh username@server and I logged in without problems.
However, I cannot log in with the graphical interface.
This is the output of the command:
1)
grep username / etc / {passwd, group, shadow, gshadow}

/ etc / passwd: username: x: 1001: 1001: ,,,: / home / username: / bin / bash
/ etc / group: username: x: 1001:
/ etc / shadow: username: pamltsp :::::::
/ etc / gshadow: username: * ::

2)
login failed

3) 
yes both

4)
yes as i said

Maybe the problem is related to the login window?
Thank you

**Comment by alkisg** [comment]
The Ctrl+Alt+F2 terminal, and the login window, both use the same mechanism: pamltsp.

When you tried in Ctrl+Alt+F2, did you see ".Login failed." with dots, or without them?
LTSP adds the dots to show that it's in control of PAM; if you don't see the bots, pamltsp isn't properly registered.

**Comment by Chwbecca** [comment]
OK understood.
Then the precise message after entering the password is:

.Login incorrect.

Login incorrect

There is a blank line between the two Login incorrect

Oh I forgot: we already have a fully functional kubuntu 20 client iso. We have problems with this new kali linux iso

**Comment by alkisg** [comment, marked as answer]
I just tested with https://cdimage.kali.org/kali-images/kali-weekly/kali-linux-2021-W42-live-amd64.iso and I didn't see any issues.
Is there something customized in your setup that breaks it?

If you want, come in the chat for live support: https://ltsp.org/advanced/chat-room/

**Reply by Chwbecca** [reply]
I don't know honestly, but in the end I believe we will use it like this, doing the same thing done for the root user.
In any case, this ISO must be used by a few people among the staff.
Thanks for the support.
